kornia/augmentation/utils/helpers.py

```diff
@@ -100,7 +102,12 @@ def _transform_input(input: Tensor) -> Tensor:
         input = input.unsqueeze(0)
 
     if len(input.shape) == 3:
-        input = input.unsqueeze(0)
+        if batch_size == input.shape[0]:
```

Comment:
I believe this can be tricky, but if we go with it only for masks may it be fine -- same comment for the _transform_output_shape

Reply:
This function looks very weird. Probably having a `transform_input_by_tensor(input: Tensor, reference_tensor: Tensor)`?

Reply:
yeah, will try to do it on Sunday

Reply:
Do we have the reference tensor in the `params`? 

What I used is the forward input shape from it

![image](https://github.com/kornia/kornia/assets/20444345/408dfd2f-2df3-406b-980f-615e906e0cea)